util/compress.go

```diff
@@ -0,0 +1,99 @@
+package util
+
+import (
+	"compress/gzip"
+	"github.com/itchio/go-brotli/enc"
+	"io"
+	"strings"
+)
+
+type CompressionType int
+
+const (
+	CompressionTypeNone    CompressionType = 0
+	CompressionTypeGzip    CompressionType = 1
+	CompressionTypeBrotli  CompressionType = 2
+)
+
+func NewGzipDecodingReader(body io.ReadCloser) (io.ReadCloser, error) {
+	gzReader, err := gzip.NewReader(body)
+	if (err != nil) {
+		return nil, err
+	}
+	return gzReader, nil
+}
+
+func NewBrotliEncodingWriter(w io.Writer) io.Writer {
+	return enc.NewBrotliWriter(w, &enc.BrotliWriterOptions{
+		Quality: 0,
+		LGWin:   0,
+	})
+}
+
+func NewGzipEncodingWriter(w io.Writer) io.Writer {
+	return gzip.NewWriter(w)
```

Comment:
Likewise, use  gzip.NewWriterLevel() and allow the integer value to be configured.